node/network/pov-distribution/src/lib.rs

```diff
@@ -296,6 +296,42 @@ async fn distribute_to_awaiting(
 	metrics.on_pov_distributed();
 }
 
+/// Connect to relevant validators in case we are not already.
+async fn connect_to_relevant_validators(
+	connection_requests: &mut validator_discovery::ConnectionRequests,
+	ctx: &mut impl SubsystemContext<Message = PoVDistributionMessage>,
+	relay_parent: Hash,
+	descriptor: &CandidateDescriptor,
+) {
+	if let Ok(Some(relevant_validators)) =
+		determine_relevant_validators(ctx, relay_parent, descriptor.para_id).await
+	{
+		// We only need one connection request per (relay_parent, para_id)
+		// so here we take this shortcut to avoid calling `connect_to_validators`
+		// more than once.
+		if !connection_requests.contains_request(&relay_parent) {
```

Comment:
I am having trouble understanding, what would happen if this function were called twice with the same `relay_parent` and two different descriptors?

Reply:
Good point. Although this code wasn't introduced in this PR, the comment says it should be unique per (relay_parent, para_id), but the only handles relay_parent.

cc @montekki #1990

Reply:
If you check the code where this is called, you see that the the connection_requests are per `pov_hash`.

Reply:
Wait, is it? `state.connection_requests` doesn't know about `pov_hash` and if we call it for two different `pov_hashes` with the same `relay_parent`, it will not issue the second request, or?

Reply:
Ahh my comment was dumb 🤦

I had overseen that connection requests is global to the state. 

The assumption was probably that we are only assigned to one parachain per relay parent?

Reply:
Do I understand it right that this is a legit problem and requires an issue?

Reply:
> Do I understand it right that this is a legit problem and requires an issue?

Don't know, the following assumption seems legit to me:

> The assumption was probably that we are only assigned to one parachain per relay parent?

Although, the less assumptions we make, the better. So if that assumption is easy to get rid of, I think that would be a good idea.

Reply:
Depends on whether this assumption is wrong
> The assumption was probably that we are only assigned to one parachain per relay parent?

If it is wrong, we'd probably need to change https://github.com/paritytech/polkadot/blob/57eb9d10d874a27f2661523ce4b533aa690589c3/node/subsystem-util/src/validator_discovery.rs#L176
to map from `(Hash, ParaId)` instead of `Hash`.